nixos/tests/cloud-init.nix

```diff
@@ -51,29 +51,33 @@ in makeTest {
     networking.hostName = "";
   };
   testScript = ''
-    machine.wait_for_unit("cloud-init.service")
-    machine.succeed("cat /tmp/cloudinit-write-file | grep -q 'cloudinit'")
+    unnamed.wait_for_unit("cloud-init.service")
+
+    # To wait until cloud-init terminates its run
+    unnamed.wait_until_succeeds("ls /run/cloud-init/result.json")
```

Comment:
This is a bit hacky. I'd really prefer if we could wait for the the service that's supposed to be creating this. Is this part of `cloud-init.service` (On which we already wait in L54), or what other unit handles the `write_files` part?

Maybe we just need to wait on another one of the services? `cloud-final.service` maybe, or just the entire `cloud-config.target`?

Reply:
This could work, but I'm not a cloud-init pro. I just wanted to fix a crashing test script and maaaaybe learn a bit of cloud-init?

Reply:
If you want, you can take a look at `nixos/modules/services/system/cloud-init.nix`, and how we invoke with various parameters in different units, then see where the `write_files` modules is handled inside cloud-init.

It's probably easier to just wait on `cloud-config.target`, which should include all of them (I think)?

Reply:
It looks like cloud-config.target includes write_files. I'm launching 200 iterations of the unit test with timeout of 120 seconds (it usually completes in 70 seconds on my machine) to test it. I guess I'll report results the next morning!

Reply:
Oh no, I've miscalculated the timeout? it timed out..

Reply:
Oh no, it looks like the test became more flaky. I guess I'll have to order after cloud-final - it includes things not included in cloud-config.target!

Reply:
Ok, after ordering the test after `cloud-final.service` it looks like everything's fine. I'll leave it to run overnight to see if there are any failures. If they are, I shall be greeted with a huge error on my console.

Reply:
Yep, would be better to be able to wait on an unit.